static/js/services/contact-view-model-generator.js

```diff
@@ -147,7 +147,25 @@ angular.module('inboxServices').factory('ContactViewModelGenerator',
       return DB().query('medic-client/contacts_by_parent', {
         key: contactId,
         include_docs: true
-      });
+      })
+        .then(function(childrenResponse) {
+          var ids = childrenResponse.rows.map(function(child) {
+            return child.doc.contact && child.doc.contact._id;
+          });
+          return DB().query('medic-client/doc_summaries_by_id', { keys: ids })
```

Comment:
I like that pattern too, unfortunately that means `childrenResponse` is then out of scope so we can't return it at the bottom (line 166).